### PR TITLE
Order details > edit product: fix dismissing a bottom sheet (product type and bottom sheet menu)

### LINKED ISSUE #2701: Order details > edit product: dismissing a bottom sheet (product type or bottom sheet menu) brings down the product form modal
**Describe the bug**

If opening a product modally (e.g. from an order) and not selecting one of the options from the bottom sheet in product details (e.g. updating the product type, or the menu at the bottom), tapping outside of the bottom sheet dismisses not only the bottom sheet but also the product details screen.

**To Reproduce**
- Go to the Orders tab
- Tap on an order
- Tap on a product that is editable
- Tap on the "+ Add more details" CTA at the bottom or the product type row (if editable)
- Tap outside of the bottom sheet to dismiss the sheet --> the whole product details is dismissed

**Screenshots**

simulator

**Expected behavior**

Only the bottom sheet is dismissed.

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: iPhone 11 simulator
- iOS version: iOS 13.5
- WooCommerce iOS version: 4.8

**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
</details>

### PR DESCRIPTION
**Pull request by jaclync**
Fixes #2701 

## Changes

- Added a default empty implementation to `BottomSheetListSelectorPresenter`'s `onDismiss` parameter, because it's only useful when tapping on each row doesn't trigger navigation actions and requires manual dismissal (e.g. in Products tab sort bottom sheet)
- In `ProductFormViewController`, updated to use `BottomSheetListSelectorPresenter` for "Add more details" bottom sheet and removed the `onDismiss` implementation that dismisses the view controller again

### Opening product details from an order

- Go to the Orders tab
- Tap on an order
- Tap on a product that is editable
- Tap on the "+ Add more details" CTA at the bottom
- Tap outside of the bottom sheet to dismiss the sheet --> only the bottom sheet is dismissed
- Tap on the product type row
- Tap outside of the bottom sheet to dismiss the sheet --> only the bottom sheet is dismissed

### Opening product details from the products tab

- Go to the Products tab
- Tap on a simple product
- Tap on the "+ Add more details" CTA at the bottom
- Tap outside of the bottom sheet to dismiss the sheet --> only the bottom sheet is dismissed
- Tap on the product type row
- Tap outside of the bottom sheet to dismiss the sheet --> only the bottom sheet is dismissed

## Example screenshots

before | after
-- | --
simulator | simulator

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/BottomSheet/ListSelector/BottomSheetListSelectorPresenter.swift [hunk 1]
```diff
@@ -6,9 +6,11 @@ import WordPressUI
 final class BottomSheetListSelectorPresenter<Command: BottomSheetListSelectorCommand> {
     private let bottomSheetChildViewController: DrawerPresentableViewController
 
+    /// - Notable Parameters:
+    ///   - onDismiss: Called when the bottom sheet is dismissed. Useful when tapping on each bottom sheet row does not trigger navigation changes.
     init(viewProperties: BottomSheetListSelectorViewProperties,
          command: Command,
-         onDismiss: @escaping (_ selected: Command.Model?) -> Void) {
+         onDismiss: @escaping (_ selected: Command.Model?) -> Void = { _ in }) {
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/BottomSheet/ListSelector/BottomSheetListSelectorPresenter.swift]: [nit] I wonder if making it optional and having `nil` as a default value would be more clear for the reader 🤔

**Review by Ecarrion — APPROVED** [review]
Great fix! 🚀

File: WooCommerce/Classes/ViewRelated/BottomSheet/ListSelector/BottomSheetListSelectorPresenter.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/BottomSheet/ListSelector/BottomSheetListSelectorPresenter.swift]: good call! updated here: https://github.com/woocommerce/woocommerce-ios/pull/2702/commits/56a01e69b97d73e37bd49d066ca23f0d9bc25485

**Comment by jaclync** [comment]
Merging the PR now, thanks for the suggestion Ernesto! I also had 2 commits that fixed `develop` build errors: https://github.com/woocommerce/woocommerce-ios/pull/2702/commits/d1f7f54ee0c23743a7e9ba1241bf32712578e10a https://github.com/woocommerce/woocommerce-ios/pull/2702/commits/09df6d078942d910433234e3eeb929af605e5263

**Pull request merged**